pkg/plugin/plugin.go

```diff
@@ -7,49 +7,37 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/openshift/openshift-azure/pkg/api"
+	pluginapi "github.com/openshift/openshift-azure/pkg/api/plugin/api"
 	"github.com/openshift/openshift-azure/pkg/arm"
 	"github.com/openshift/openshift-azure/pkg/cluster"
 	"github.com/openshift/openshift-azure/pkg/config"
 )
 
 type plugin struct {
-	log             *logrus.Entry
-	config          api.PluginConfig
-	clusterUpgrader cluster.Upgrader
-	armGenerator    arm.Generator
-	configGenerator config.Generator
-	apiValidator    *api.Validator
-	adminValidator  *api.Validator
+	log                     *logrus.Entry
+	config                  api.PluginConfig
+	clusterUpgrader         cluster.Upgrader
+	armGenerator            arm.Generator
+	configGenerator         config.Generator
+	apiValidator            *api.Validator
+	adminValidator          *api.Validator
+	pluginTemplateValidator *api.Validator
```

Comment:
2 of these is bad, 3 is worse.  Do these different validations really want to be off the same api.Validator base type?

Reply:
our current Validator is super simple. If it becomes different - we can split. 
What the alternative? Create a separate base type?

Reply:
yes - what does moving in the direction of separate file and base type for each separate api package type validated look like?

Reply:
I might do it in the second PR as it will make this even more complicated after test. Will raise Issue and assign

Reply:
follow-up: https://github.com/openshift/openshift-azure/issues/967